subsys/rpmsg_service/rpmsg_backend.c

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright (c) 2021, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "rpmsg_backend.h"
+
+#include <zephyr.h>
+#include <drivers/ipm.h>
+#include <device.h>
+#include <logging/log.h>
+
+#include <openamp/open_amp.h>
+#include <metal/device.h>
+
+#define LOG_LEVEL LOG_LEVEL_INFO
+#define LOG_MODULE_NAME rpmsg_backend
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+/* Configuration defines */
+#if !DT_HAS_CHOSEN(zephyr_ipc_shm)
+#error "Module requires definition of shared memory for rpmsg"
+#endif
+
+#define MASTER IS_ENABLED(CONFIG_RPMSG_SERVICE_MODE_MASTER)
+
+#if MASTER
+#define VIRTQUEUE_ID 0
+#define RPMSG_ROLE RPMSG_MASTER
+#else
+#define VIRTQUEUE_ID 1
+#define RPMSG_ROLE RPMSG_REMOTE
+#endif
+
+/* Configuration defines */
+
+#define VRING_COUNT		    2
+#define VRING_RX_ADDRESS	(VDEV_START_ADDR + SHM_SIZE - VDEV_STATUS_SIZE)
+#define VRING_TX_ADDRESS	(VDEV_START_ADDR + SHM_SIZE)
+#define VRING_ALIGNMENT		4
+#define VRING_SIZE		    16
+
+#define IPM_WORK_QUEUE_STACK_SIZE 2048
```

Comment:
the size of the stack seems very high to me

Reply:
Stack size was increased to this value to solve stack overflow problems. Stack overflow was caused by long IPC messages placed in the stack. I think making this value configurable to meet application needs would be better. What do you think?

Reply:
Yes could be nice